### ISSUE TITLE
`no-warning-comments` break on hyphen char

### ISSUE THREAD

**Issue by pascalduez** [eslint/eslint#11471]
**Tell us about your environment**

* **ESLint Version:** `5.15.0`
* **Node Version:** `11.10.1`
* **npm Version:** `6.8.0`

**What parser (default, Babel-ESLint, etc.) are you using?**

`babel-eslint`

**Please show your full configuration:**

```json
{
  "parser": "babel-eslint",
  "plugins": [
    "flowtype"
  ],
  "rules": {
    "flowtype/no-weak-types": 1,
    "no-warning-comments":  [2, { "terms": ["flow-typed signature", "flow-typed version"], "location": "start"}]
  }
}
```

  **What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Upgraded ESlint from `5.7.0` to `5.15.0`. 

```bash
eslint **/**/test_* --quiet
```

**What did you expect to happen?**

Linting :unicorn: 

**What actually happened? Please include the actual, raw output from ESLint.**

The rule choke on the hypen in `"flow-typed"`. 
```json
"no-warning-comments":  [2, { "terms": ["flow-typed signature", "flow-typed version"], "location": "start"}]
````

```
SyntaxError: Error while loading rule 'no-warning-comments': Invalid regular expression: /^\s*flow\-typed version\b/: Invalid escape
    at new RegExp (<anonymous>)
    at convertToRegExp (flow-typed/definitions/node_modules/eslint/lib/rules/no-warning-comments.js:98:24)
    at Array.map (<anonymous>)
    at Object.create (flow-typed/definitions/node_modules/eslint/lib/rules/no-warning-comments.js:109:45)
    at createRuleListeners (flow-typed/definitions/node_modules/eslint/lib/linter.js:578:21)
    at Object.keys.forEach.ruleId (flow-typed/definitions/node_modules/eslint/lib/linter.js:732:31)
    at Array.forEach (<anonymous>)
    at runRules (flow-typed/definitions/node_modules/eslint/lib/linter.js:690:34)
    at Linter._verifyWithoutProcessors (flow-typed/definitions/node_modules/eslint/lib/linter.js:894:31)
    at preprocess.map.textBlock (flow-typed/definitions/node_modules/eslint/lib/linter.js:943:35)
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. I can reproduce this issue.
